app/views/ReferenceSelectView/ReferenceSelectView.jsx

```diff
@@ -449,6 +446,54 @@ class ReferenceSelectView extends React.Component {
   }
 }
 
-export default connectRedux(documentActions, selectionActions)(
+function mapState(state, ownProps) {
+  const {
+    route: {params, path, searchString}
+  } = ownProps
+
+  const collection =
+    params.collection === Constants.MEDIA_COLLECTION_SCHEMA.slug
+      ? Constants.MEDIA_COLLECTION_SCHEMA
+      : state.app.config.api.collections.find(collection => {
+          return collection.slug === params.collection
+        })
+
+  const isNewDocument = /\/new\/?/.test(path) && !params.documentId
```

Comment:
Is this a good check? It correctly matches `/articles/new`, but also matches `/newyork/12345`.